src/compat.jl

```diff
@@ -0,0 +1,3 @@
+@static if VERSION < v"1.7-"
```

Comment:
Just drop a support for v1.6...?

Reply:
I'm not sure if I can get YaoCompiler working again this weekend, so I think I'll probably try to keep the first version compatible with 1.6 then deprecate it